src/nvim/diff.c

```diff
@@ -2212,6 +2975,10 @@ int diffopt_changed(void)
       } else {
         return FAIL;
       }
+    } else if ((STRNCMP(p, "linematch:", 10) == 0) && ascii_isdigit(p[11])) {
```

Comment:
Does this properly handle `linematch:1text`?

Reply:
fixed, assuming this is what you meant:
https://github.com/neovim/neovim/blob/ea46c57d8e21ec123e2665134592fa8ecd921d66/src/nvim/diff.c#L2982-L2989